api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -685,15 +697,33 @@ private boolean userHasEverLoggedIn(
     return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
   }
 
-  private void updateProfileForUser(DbUser user, Profile updatedProfile, Profile previousProfile) {
+  public ResponseEntity<Void> adminUpdateProfile(DbUser user, Profile updatedProfile) {
+    // Save current profile for audit trail.
+    final Profile previousProfile = profileService.getProfile(user);
+
     validateUpdatedProfile(updatedProfile, previousProfile);
 
-    if (!userProvider.get().getGivenName().equalsIgnoreCase(updatedProfile.getGivenName())
-        || !userProvider.get().getFamilyName().equalsIgnoreCase(updatedProfile.getFamilyName())) {
-      userService.setDataUseAgreementNameOutOfDate(
-          updatedProfile.getGivenName(), updatedProfile.getFamilyName());
+    profileService.adminUpdateProfile(updatedProfile);
```

Comment:
adminUpdateProfile uses a mostly-different path here, so as not to affect other endpoints much.  RW-5104 will unify them.

Reply:
Can you please write up a 1-2 page plan for how the calls are going to work and where the responsibilities are going to lie now?